pkg/controllers/vdb/sandboxsubcluster_reconciler.go

```diff
@@ -153,27 +158,86 @@ func (s *SandboxSubclusterReconciler) sandboxSubclusters(ctx context.Context) (c
 
 // executeSandboxCommand will call sandbox API in vclusterOps, create/update sandbox config maps,
 // and update sandbox status in vdb
-func (s *SandboxSubclusterReconciler) executeSandboxCommand(ctx context.Context, scSbMap map[string]string) error {
+func (s *SandboxSubclusterReconciler) executeSandboxCommand(ctx context.Context, scSbMap map[string]string) (ctrl.Result, error) {
 	succeedSbScMap := make(map[string][]string)
 	seenSandboxes := make(map[string]any)
-	for sc, sb := range scSbMap {
-		err := s.sandboxSubcluster(ctx, sc, sb)
-		if err != nil {
-			// when one subcluster failed to be sandboxed, update sandbox status and return error
-			return errors.Join(err, s.updateSandboxStatus(ctx, succeedSbScMap))
-		}
-		succeedSbScMap[sb] = append(succeedSbScMap[sb], sc)
-		// create/update a sandbox config map
-		if _, ok := seenSandboxes[sb]; !ok {
-			err = s.checkSandboxConfigMap(ctx, sb)
-			if err != nil {
-				// when creating/updating sandbox config map failed, update sandbox status and return error
-				return errors.Join(err, s.updateSandboxStatus(ctx, succeedSbScMap))
+
+	// We can simply loop over the scSbMap and sandbox each subcluster. However,
+	// we want to sandbox in a determinstic order because the first subcluster
+	// in a sandbox is the primary.
+	for i := range s.Vdb.Spec.Sandboxes {
+		vdbSb := &s.Vdb.Spec.Sandboxes[i]
+		for j := range vdbSb.Subclusters {
+			sc := vdbSb.Subclusters[j].Name
+			sb, found := scSbMap[sc]
+			if !found {
+				// assume it is already sandboxed
+				continue
 			}
+			// The first subcluster in a sandbox turns into a primary. Set
+			// state in the vdb to indicate that.
+			if j == 0 {
+				_, err := vk8s.UpdateVDBWithRetry(ctx, s.VRec, s.Vdb, func() (bool, error) {
+					scMap := s.Vdb.GenSubclusterMap()
+					vdbSc, found := scMap[sc]
+					if !found {
+						return false, fmt.Errorf("subcluster %q missing in vdb %q", sc, s.Vdb.Name)
+					}
+					vdbSc.Type = vapi.SandboxPrimarySubcluster
```

Comment:
Why do we need to update this? Isn't this set by the user in the spec? I think we just need to sandbox the subcluster with "sandboxprimary" first.

Reply:
It makes it easier to sandbox if there is one less thing the user needs to update. Plus, if we needed the user to change the type of the subcluster they want as the primary, we would need to ensure we sandbox that subcluster first. I think it's easier if we assume the first subcluster listed in the sandbox becomes the primary.

Reply:
I think we can improve this in VER-94031